docs/proposal/2022-06-14-configuration-bindings.md

```diff
@@ -0,0 +1,647 @@
+
+    Related issues:
+      - [Multiple Slack Channels](https://github.com/infracloudio/botkube/issues/250)
+
+### Non-goal
+
+Those are not the goals of this proposal. However, we should be able to implement them later.
+
+- [Notify groups](https://github.com/infracloudio/botkube/issues/323)
+- [Group error messages to thread](https://github.com/infracloudio/botkube/issues/545)
+- [Customizable Messages](https://github.com/infracloudio/botkube/issues/434)
+- [Support for default cluster and namespace on any Slack channels](https://github.com/infracloudio/botkube/issues/421)
+
+## Proposal
+
+### Terminology
+
+| Name           | Description                                                                                                                                                            |
+|----------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Executors      | Executes BotKube or `kubectl` command and sends back the result to the Bot. In the future, new executors might be added. For example, `helm`, `argo`, `istioctl`, etc. |
+| Sources        | Provides domain specific notification. For example, Kubernetes events, security events (Sysdig), metrics (Prometheus), and similar.                                    |
+| Communications | Describes both Bot and Sink.                                                                                                                                           |
+| Bot            | Bi-directional communication such as Slack, Discord, Mattermost.                                                                                                       |
+| Sink           | Unidirectional communication, such as Elasticsearch or Webhook. It may receive events from Sources only.                                                               |
+
+![](../assets/ideation-view.png#gh-light-mode-only)
+![](../assets/ideation-view-dark.png#gh-dark-mode-only)
+
+### New syntax
+
+This section describes the necessary changes in the syntax. **It's not backward compatible.**
+
+1. Sources
+    <table>
+    <tr>
+    <td> Before </td> <td> After </td>
+    </tr>
+    <tr>
+    <td>
+
+    ```yaml
+
+
+
+
+
+
+
+
+
+    resources:
+      - name: v1/pods
+        namespaces:
+          include: ["istio-system"]
+        events: ["error"]
+      - name: networking.istio.io/v1alpha3/VirtualServices
+        namespaces:
+          include: ["@all"]
+        events: ["error"]
+
+    # Recommendations about the best practices
+    recommendations: true
+
+
+
+
+
+    ```
+
+    </td>
+    <td>
+
+    ```yaml
+    sources:
+      - name: default
+        kubernetes:
+          # New 'namespace' property - allows Namespace restriction
+          # It can be overridden in the nested level.
+          namespace:
+            include: ["@all"]
+          resources:
+            - name: v1/pods
```

Comment:
what about `updateSetting` ? do you plan to keep this feature?
it's great, I don't think it would be a good idea to remove it :)

Reply:
maybe better to specify this :)